packages/validator/src/services/attestation.ts

```diff
@@ -163,11 +163,10 @@ export class AttestationService {
     const afterBlockDelayMs = (1000 * this.clock.secondsPerSlot) / (this.opts?.afterBlockDelaySlotFraction ?? 6);
     await sleep(Math.min(msToOneThirdSlot, afterBlockDelayMs));
 
-    this.metrics?.attesterStepCallPublishAttestation.observe(this.clock.secFromSlot(slot + 1 / 3));
-
     // Step 2. Publish all `Attestations` in one go
     try {
       await this.api.beacon.submitPoolAttestations(signedAttestations);
+      this.metrics?.attesterStepCallPublishAttestation.observe(this.clock.secFromSlot(slot + 1 / 3));
```

Comment:
Agree with @wemeetagain the metric name clearly states "before call to publish attestation". You can already get the metric you want by summing `attesterStepCallPublishAttestation + vc rest request time {routeId=submitPoolAttestations}`.

If you want to be more granular metrics, those should be added in the beacon chain not here

Reply:
If your goal is to measure how busy the beacon node is (potentially due to i/o lag), just use the api request time metric